code/modules/projectiles/guns/special/blastcannon.dm

```diff
@@ -84,7 +90,11 @@
 	return ..()
 
 /obj/item/gun/blastcannon/update_icon_state()
-	icon_state = "[bomb ? "loaded" : "empty"]_[base_icon_state]"
+	icon_state = "[base_icon_state]_[bomb ? "loaded" : "empty"]"
+	inhand_icon_state = icon_state
+	if(ismob(loc))
+		var/mob/cannoneer = loc
+		cannoneer.update_inv_hands()
```

Comment:
theres an element/component for this, called something like update icon update onmob

Reply:
good catch!